src/pushsource/_impl/source.pyi

```diff
@@ -0,0 +1,42 @@
+from collections import Mapping, Callable, Iterator
+from types import TracebackType
+from typing import Type, TypeVar, Any
+
+from pushsource._impl.model.base import PushItem_co
+
+SourceFactory = Callable[[], "Source"]
```

Comment:
I would rather not require that a SourceFactory has to return a type of Source or a subclass.  At the moment, you can actually return any iterable of push items and some tests take advantage of this, because it's an easy way to set up a test if you just have a hardcoded list of items you want to return.

Reply:
How about this instead (incorporating the fact that immutable `Iterable` is covariant and thus doesn't need `PushItem_co`):

```python
PushItemIterator = Callable[[], Iterable[PushItem]]
```